packages/dev/core/src/Rendering/objectRenderer.ts

```diff
@@ -0,0 +1,684 @@
+    /**
+     * Objects will be rendered every frame and is recommended for dynamic contents.
+     */
+    public static readonly REFRESHRATE_RENDER_ONEVERYFRAME: number = 1;
+    /**
+     * Objects will be rendered every 2 frames which could be enough if your dynamic objects are not
+     * the central point of your effect and can save a lot of performances.
+     */
+    public static readonly REFRESHRATE_RENDER_ONEVERYTWOFRAMES: number = 2;
+
+    /**
+     * Use this predicate to dynamically define the list of mesh you want to render.
+     * If set, the renderList property will be overwritten.
+     */
+    public renderListPredicate: (AbstractMesh: AbstractMesh) => boolean;
+
+    private _renderList: Nullable<Array<AbstractMesh>>;
+    private _unObserveRenderList: Nullable<() => void> = null;
+
+    /**
+     * Use this list to define the list of mesh you want to render.
+     */
+    public get renderList(): Nullable<Array<AbstractMesh>> {
+        return this._renderList;
+    }
+
+    public set renderList(value: Nullable<Array<AbstractMesh>>) {
+        if (this._renderList === value) {
+            return;
+        }
+        if (this._unObserveRenderList) {
+            this._unObserveRenderList();
+            this._unObserveRenderList = null;
+        }
+
+        if (value) {
+            this._unObserveRenderList = _ObserveArray(value, this._renderListHasChanged);
+        }
+
+        this._renderList = value;
+    }
+
+    private _renderListHasChanged = (_functionName: String, previousLength: number) => {
+        const newLength = this._renderList ? this._renderList.length : 0;
+        if ((previousLength === 0 && newLength > 0) || newLength === 0) {
+            this._scene.meshes.forEach((mesh) => {
+                mesh._markSubMeshesAsLightDirty();
+            });
+        }
+    };
+
+    /**
+     * Define the list of particle systems to render. If not provided, will render all the particle systems of the scene.
+     * Note that the particle systems are rendered only if renderParticles is set to true.
+     */
+    public particleSystemList: Nullable<Array<IParticleSystem>> = null;
+
+    /**
+     * Use this function to overload the renderList array at rendering time.
+     * Return null to render with the current renderList, else return the list of meshes to use for rendering.
+     * For 2DArray, layerOrFace is the index of the layer that is going to be rendered, else it is the faceIndex of
+     * the cube (if the RTT is a cube, else layerOrFace=0).
+     * The renderList passed to the function is the current render list (the one that will be used if the function returns null).
+     * The length of this list is passed through renderListLength: don't use renderList.length directly because the array can
+     * hold dummy elements!
+     */
+    public getCustomRenderList: (layerOrFace: number, renderList: Nullable<Immutable<Array<AbstractMesh>>>, renderListLength: number) => Nullable<Array<AbstractMesh>>;

```

Comment:
Seems like this whole thing should be `Nullable`?
```suggestion
    public getCustomRenderList: Nullable<(layerOrFace: number, renderList: Nullable<Immutable<Array<AbstractMesh>>>, renderListLength: number) => Nullable<Array<AbstractMesh>>> = null;

```

Reply:
Same thing for the other funcs later in this file

Reply:
I can't do that, it would be a breaking change from what `RenderTargetTexture` is currently doing.

Note that I just moved code from `RenderTargetTexture` to `ObjectRenderer` and tried to change as few things as possible to not introduce breaking changes.

Reply:
We finally decided with @sebavan that it should not break any code, so I did the change.